modules/planning/scenarios/scenario_manager.cc

```diff
@@ -530,17 +531,25 @@ ScenarioConfig::ScenarioType ScenarioManager::SelectParkAndGoScenario(
       common::VehicleStateProvider::Instance()->vehicle_state();
   auto adc_point = common::util::MakePointENU(
       vehicle_state.x(), vehicle_state.y(), vehicle_state.z());
+  // TODO(SHU) might consider gear == GEAR_PARKING
+  double adc_speed =
+      common::VehicleStateProvider::Instance()->linear_velocity();
   double s = 0.0;
   double l = 0.0;
+  const double max_abs_speed_when_stopped =
+      common::VehicleConfigHelper::Instance()
+          ->GetConfig()
+          .vehicle_param()
+          .max_abs_speed_when_stopped();
+
   hdmap::LaneInfoConstPtr lane;
-  if (HDMapUtil::BaseMap().GetNearestLaneWithHeading(
-          adc_point, 2.0, vehicle_state.heading(), M_PI / 3.0, &lane, &s, &l) !=
-      0) {
+  // if vehicle is static and (off-lane or not on city_driving lane)
+  if (std::fabs(adc_speed) < max_abs_speed_when_stopped &&
+      (HDMapUtil::BaseMap().GetNearestLaneWithHeading(
```

Comment:
add another if here, since lane_type check should be inside when speed check is true.